Moved the communicator to a library project

Edith-D opened the pull request:
Create an Interface for the Communicator

Edith-D committed:
Moved the communicator to a library project and create an Interface for the communicator.

Edith-D committed:
Create IMessage and IResponse in Share Project

raksanyi commented on Shared/InternalsVisibleTo.cs:
```diff
@@ -0,0 +1,3 @@
+﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("WPFClient")]
```
Keep in mind, this at the moment is only needed because we don't have DI container setups yet. In the future this will have to be removed. The Communicator class should be internal, and the clients will only see the ICommunicator interface.

raksanyi commented on CorEstafette.sln:
```diff
@@ -5,9 +5,11 @@ VisualStudioVersion = 16.0.809.6
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebClient", "WebClient\WebClient.csproj", "{1B1EEC10-7DF8-484C-9571-F294592DD05B}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "WebClient", "WebClient\WebClient.csproj", "{1B1EEC10-7DF8-484C-9571-F294592DD05B}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Shared", "Shared\Shared.csproj", "{1F4D49A6-3246-47F1-A0EC-B282B6CC3D20}"
```
This can be renamed to Communicator.csproj, and put everything that the published nuget package will contain in it (Communicator, ICommunicator, IMessage, IResponse, etc). And then let the signalr service and the clients reference it (the service can get the InternalsVisibleTo is needed).

Edith-D committed:
Change Shared Project to SignalRCommunicator

Edith-D reviewed (commented):
- In Response 
      1. Added the full contructor calling empty constructor to enable automatic serialisation and deserialisation from ClientModelView
- Move Share project to be SignalRCommunicator
- Subscribe and unsubscribe
  1. Return a Response from the hub instead of a string, state stored in the Content of the response
  2. Catch the Response in the communicator
  3. Add a timeout check in communicator
  4. Return a Response to call from the Communicator.  Success of fail text found in the Content of the Response.  
  5. 3 possibles messages, Success, Timeout or already subscribe or can't unsubscribe not subscribe
- In the communicator, Add a generated userId to differentiate user from different running UI
- Clean up code.

Edith-D committed:
OnPublish receives a Message
In Communicator : Changed the output of message to see from which topic and who is sending message.
In TestHub : Sends bakc a message instead of the 2 strings
Cleaned up code.

Edith-D committed:
Resplaced Response to IResponse when possible
Rename TestHub to SignalRHub and made the changes where appropriate
Fixed the WhenAny conditions so that it does stop after first task is done.

Edith-D committed:
Merge branch 'master' into MainWindow
